datafusion/core/src/execution/session_state.rs

```diff
@@ -1144,7 +1146,9 @@ impl SessionStateBuilder {
         mut self,
         expr_planners: Vec<Arc<dyn ExprPlanner>>,
     ) -> Self {
-        self.expr_planners = Some(expr_planners);
+        if self.expr_planners.is_none() {
```

Comment:
By putting these checks here I think the API will be quite confusing to use -- it means that calling `with_expr_planners` and the other methods below will only set the fields if there is no existing field set and silently ignore the argument if there is already planners specified